illumina_somaticVariants.pm

```diff
@@ -34,8 +34,11 @@ sub parseSamples {
         if($opt{SET_HAS_METADATA} eq "yes"){
             print "Reading Tumour/Reference names from metadata\n\n";
             my $metadata = metadataParse($opt{OUTPUT_DIR});
-            push(@{$somatic_samples{$cpct_name}{"ref"}}, $metadata->{'ref_sample'});
-            push(@{$somatic_samples{$cpct_name}{"tumor"}}, $metadata->{'tumor_sample'});
+            if($sample =~ $metadata->{'ref_sample'}){
+              push(@{$somatic_samples{$cpct_name}{"ref"}}, $metadata->{'ref_sample'});
+            } elsif ($sample =~ $metadata->{'tumor_sample'}){
+              push(@{$somatic_samples{$cpct_name}{"tumor"}}, $metadata->{'tumor_sample'});
+            }
```

Comment:
If we are using metadata and it does not match the sample name, I'd probably expect an error/warning in an `else` clause.